packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -63,27 +92,94 @@ export class OptimisticOracleRequest {
     this.resolvedValue = data.resolvedValue;
     this.resolveTx = data.resolveTx;
     this.disputeTx = data.disputeTx;
+    this.blockNumber = data.blockNumber;
+    this.transactionIndex = data.transactionIndex;
+  }
+
+  getId(): string {
+    return calculateRequestId(this.body, this.identifier, this.timestamp, this.requester);
+  }
+
+  setProposer(proposer: string): void {
+    this.proposer = proposer;
+  }
+
+  setProposedValue(proposedValue: number | boolean): void {
```

Comment:
Why do we have all these mutable functions? Would it be easier just to set the props directly? Why did you decide to move away from immutability?